### PR TITLE
Fix/1818  switch app folder advanced

### PR DESCRIPTION
**Pull request by Skylsmoi**
https://github.com/tracim/tracim/issues/1818

## Checkpoints

- [x] Code is clear enoug
- [x] If there are FIXME in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODO, NOTE or HACK in code, date and developer initials are present
- [x] Automated tests have been written (covering feature or non regression), or *at least* an issue has been created for test implementation

### CODE REVIEW

File: functionnal_tests/cypress/integration/app_folder/app_folder_spec.js [hunk 1]
```diff
@@ -1,34 +1,83 @@
...
+    cy.get('#appFeatureContainer > [data-cy=popinFixed].folder_advanced')
+      .find('[data-cy="popinFixed__header__button__close"]')
+      .click()
+
+    cy.getTag({selectorName: s.FOLDER_IN_LIST, params: {folderId: folder2.content_id}})
+      .find('.extandedaction.dropdown')
+      .click()
+
+    cy.getTag({selectorName: s.FOLDER_IN_LIST, params: {folderId: folder2.content_id}})
+      .find('.extandedaction.dropdown .fa.fa-pencil')
```
Comment (basilelegal) [inline functionnal_tests/cypress/integration/app_folder/app_folder_spec.js]: I think we need a data-cy for this selector, this is a generic component and we can't base our test on an icon.

File: functionnal_tests/cypress/support/db_commands.js [hunk 2]
```diff
@@ -201,6 +201,28 @@ Cypress.Commands.add('createThread', (title, workspaceId, parentId = null) => {
...
+  let data = {
+    content_type: 'folder',
+    label: title,
+    parent_id: parentId
+  }
+  cy
+    .request('POST', url, data)
+    .then(response => {
+      if (response === undefined) {
+        // FIXME -  B.L - 2019/05/03 - when we send simultaneous request to create contents we
```
Comment (basilelegal) [inline functionnal_tests/cypress/support/db_commands.js]: you are not B.L

File: functionnal_tests/cypress/integration/app_folder/app_folder_spec.js [hunk 3]
```diff
@@ -1,34 +1,83 @@
...
+      .click()
+
+    cy.get('#appFeatureContainer > [data-cy=popinFixed].folder_advanced')
+      .find('.folderAdvanced__header__title')
+      .should('contain', folder2.label)
+  })
 })
```
Comment (basilelegal) [inline functionnal_tests/cypress/integration/app_folder/app_folder_spec.js]: We need one more test to be sure, that when we close a folder, then re open it it's still opening

**Comment by Skylsmoi** [comment]
@Skylsmoi check again unchecking an app type, closing the app, reopen it and check back the type

File: functionnal_tests/cypress/integration/app_folder/app_folder_spec.js (hunk 3 above)
Comment (Skylsmoi) [reply functionnal_tests/cypress/integration/app_folder/app_folder_spec.js]: test added

File: functionnal_tests/cypress/support/db_commands.js (hunk 2 above)
Comment (Skylsmoi) [reply functionnal_tests/cypress/support/db_commands.js]: identity not usurped anymore

File: functionnal_tests/cypress/integration/app_folder/app_folder_spec.js (hunk 1 above)
Comment (Skylsmoi) [reply functionnal_tests/cypress/integration/app_folder/app_folder_spec.js]: data-cy added

File: functionnal_tests/cypress/support/generic_selector_commands.js [hunk 4]
```diff
@@ -2,12 +2,14 @@ const SELECTORS = {
...
   FOLDER_IN_LIST: 'folderInList',
-  WORKSPACE_MENU: 'workspaceMenu'
+  WORKSPACE_MENU: 'workspaceMenu',
+  CONTENT_IN_RESEARCH: 'contentInResearch'
 }
 
 const TAGS = {
   [SELECTORS.CONTENT_FRAME]: () => '[data-cy="popinFixed"]',
   [SELECTORS.CONTENT_IN_LIST]: () => `.workspace__content__fileandfolder > .content`,
+  [SELECTORS.CONTENT_IN_RESEARCH]: () => `.content__name`,
```
Comment (basilelegal) [inline functionnal_tests/cypress/support/generic_selector_commands.js]: merge conflict ?

Comment (Skylsmoi) [reply functionnal_tests/cypress/support/generic_selector_commands.js]: it is

**Pull request merged**